schema/me/append_conversation_thread.js

```diff
@@ -35,25 +48,36 @@ export default mutationWithClientMutationId({
     },
   },
   outputFields: {
-    message: {
-      type: MessagePayloadType,
-      resolve: message => message,
+    payload: {
+      type: AppendConversationThreadMutationPayload,
+      resolve: data => data,
     },
   },
   mutateAndGetPayload: ({ id, from, to, body_text }, request, { rootValue: { accessToken } }) => {
     if (!accessToken) return null
-    return gravity
-      .with(accessToken, { method: "POST" })("me/token", {
-        client_application_id: IMPULSE_APPLICATION_ID,
-      })
+    let newMessagePayload
+    return getImpulseToken(accessToken)
       .then(data => {
         return Promise.resolve(
           impulse.with(data.token, { method: "POST" })(`conversations/${id}/messages`, {
             to: ["partner@example.com"],
             from,
             body_text,
           })
-        ).then(message => message)
+        )
+      })
+      .then(payload => {
+        newMessagePayload = payload
+        return getImpulseToken(accessToken)
```

Comment:
Hmm, the token should be good for 1 minute, are you saying it takes longer than that to add a 
 message to a conversation and then re-fetch the messages?

We could possibly bump that expiration time up as well if needed.